profiling/src/timeline.rs

```diff
@@ -204,21 +204,20 @@ unsafe extern "C" fn ddog_php_prof_zend_error_observer(
     }
 
     #[cfg(zend_error_observer_80)]
-    let file = unsafe {
-        let mut len = 0;
-        let file = file as *const u8;
-        while *file.add(len) != 0 {
-            len += 1;
-        }
-        std::str::from_utf8_unchecked(std::slice::from_raw_parts(file, len)).to_string()
+    let filename = unsafe {
+        let cstr = core::ffi::CStr::from_ptr(file);
+        cstr.to_string_lossy().into_owned()
```

Comment:
I like the symmetry of the code when using `zai_str_from_cstr()` (like in the other PR) but it feels that it is just the same as this code with extra steps 😄

Reply:
Perhaps you would like this better:

```rs
    #[cfg(zend_error_observer_80)]
    let filename_str = unsafe { core::ffi::CStr::from_ptr(file) };
    #[cfg(not(zend_error_observer_80))]
    let filename_str = unsafe { zai_str_from_zstr(file.as_mut()) };
    
    let filename = filename_str.into_string_lossy().into_owned();
```

The type of `filename_str` varies which is why I didn't write it this way, but you can see how the operations are identical. The way I wrote it shows that, none of the lossy-bits are hidden away. They're doing the same thing on both branches.

Note that you can't save the `into_string_lossy()` into a variable like this because of lifetimes:

```rs
    #[cfg(zend_error_observer_80)]
    let filename_lossy = unsafe { core::ffi::CStr::from_ptr(file).into_string_lossy() };
    #[cfg(not(zend_error_observer_80))]
    let filename_lossy = unsafe { zai_str_from_zstr(file.as_mut()).into_string_lossy() };
    
    let filename = filename_str.into_owned();
```

This would have been my preferred way to write it.

Reply:
LGTM

Reply:
To clarify: you like the PR as-is or do you like this better?

```rust
    #[cfg(zend_error_observer_80)]
    let filename_str = unsafe { core::ffi::CStr::from_ptr(file) };
    #[cfg(not(zend_error_observer_80))]
    let filename_str = unsafe { zai_str_from_zstr(file.as_mut()) };
    
    let filename = filename_str.into_string_lossy().into_owned();
```

Reply:
I like the change you proposed in the comment above better